sql/scan.go

```diff
@@ -79,7 +79,9 @@ type scanNode struct {
 	kvIndex          int               // current index into the key/value pairs
 	rowIndex         int               // the index of the current row
 	colID            ColumnID          // column ID of the current key
+	valTypes         []parser.Datum    // the index key value types for the current row
 	vals             []parser.Datum    // the index key values for the current row
+	implicitValTypes []parser.Datum    // the implicit value types for unique indexes
```

Comment:
a comment about these ...Types members being necessary to handle NULL's lack of type-preservation might be good